### PR TITLE
better per stream rate limits configuration options

### PR DESCRIPTION
**Pull request by owen-d**
This PR is part one of a two phase PR moving
`max_stream_rate_bytes -> per_stream_rate_limit`
and
`max_stream_burst_rate_bytes -> per_stream_rate_limit_burst`

I'm doing this before we release 2.4 in order to move towards more intuitive field names and in the case of the burst configs, a more accurate name (the `burst` is a static value determining how far "ahead" of the rate limit a stream may be).

The followup PR will remove the deprecated field entirely (to be timed with the next `k-branch` cutting).

### CODE REVIEW

File: docs/sources/configuration/_index.md [hunk 1]
```diff
@@ -1910,6 +1910,17 @@ logs in Loki.
...
 [max_queriers_per_tenant: <int> | default = 0]
+
+# Maximum byte rate per second per stream,
+# also expressible in human readable forms (1MB, 256KB, etc).
+# CLI flag: -ingester.per-stream-rate-limit
+[per_stream_rate_limit: <string|int> | default = "3MB"]
+
+# Maximum burst bytes per stream,
+# also expressible in human readable forms (1MB, 256KB, etc).
+# This is how far over the rate limit a stream can "burst" before it's limited.
```
Comment (KMiller-Grafana) [inline docs/sources/configuration/_index.md:1921]:
```suggestion
# This is how far above the rate limit a stream can "burst" before the stream is limited.
```

**Review by KMiller-Grafana — APPROVED** [review]
I've suggested a wording improvement.  Your wording is correct and fine, and I approve the docs portion of this PR.  Consider the change, to be more concrete in word usage.

**Review by dannykopping — APPROVED** [review]
LGTM

**Comment by owen-d** [comment]
ref https://github.com/grafana/loki/issues/1544

**Pull request merged**